mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -328,14 +671,137 @@ void mav_vehicle::takeoff()
 
 void mav_vehicle::rotate(double angle_deg)
 {
+    cmd_custom cmd = cmd_custom::ROTATE;
+
+    // Check if cmd_custom::ROTATE has been sent recently
+    if (cmd_custom_timestamps.count(cmd) &&
+        !is_timedout(cmd_custom_timestamps[cmd],
+                     request_intervals_ms::rotate)) {
+        return;
+    }
+
+    // Set rotation goal
+    this->rotation_goal = math::deg2rad(angle_deg);
+
+    // We need to make sure mode is set as GUIDED in order to send the rotation
+    // command
+    set_mode(mode::GUIDED, 0);
+
+    // Set a detour waypoint on the current position
+    // send_detour_waypoint(this->global);
+
+    // Send the rotation command immediately
     send_cmd_long(MAV_CMD_CONDITION_YAW, fabs(angle_deg),
                   defaults::lookat_rot_speed_degps, -copysign(1, angle_deg), 1,
-                  0, 0, 0, 2000);
+                  0, 0, 0, 0);
+
+    // Update timestamp
+    cmd_custom_timestamps[cmd] = std::chrono::system_clock::now();
+
+    print_verbose("Sending rotation command\n");
+}
+
+bool mav_vehicle::is_rotation_active() const
+{
+    return this->rotation_active;
+}
+
+void mav_vehicle::send_mission_waypoint(global_pos_int global)
+{
+
+    double lat = double(global.lat) / 1e7;
+    double lon = double(global.lon) / 1e7;
+    double alt = double(global.alt) / 1e3 - double(this->home.alt) / 1e3;
+
+    send_mission_waypoint(lat, lon, alt);
 }
 
-void mav_vehicle::goto_waypoint(double lat, double lon, double alt)
+void mav_vehicle::send_mission_waypoint(double lat, double lon, double alt)
+{
+
+    // We need to toggle from AUTO to GUIDED in order to update the mission
+    // waypoints
+    set_mode(mode::GUIDED, 0);
+
+    // Create a waypoint to be saved into the mission_items_list
+    global_pos_int wp;
+    wp.lat = lat * 1e7;
```

Comment:
This is just a conversion from floating-point to fixed-point. I've included a comment saying that.